rust/src/nasl/builtin/ssh/libssh/mod.rs

```diff
@@ -0,0 +1,152 @@
+// SPDX-FileCopyrightText: 2023 Greenbone AG
+//
+// SPDX-License-Identifier: GPL-2.0-or-later WITH x11vnc-openssl-exception
+
+//! Defines NASL ssh and sftp functions
+
+mod channel;
+mod session;
+
+use std::net::{IpAddr, UdpSocket};
+use std::{os::fd::AsRawFd, time::Duration};
+
+use libssh_rs::{LogLevel, SshOption};
+use russh::cipher;
+use russh_keys::key;
+use tokio::sync::{Mutex, MutexGuard};
+use tracing::debug;
+
+use super::error::Result;
+use super::Ssh;
+
+pub use libssh_rs::AuthMethods;
+pub use session::SshSession;
+
+pub type SessionId = i32;
+pub type Socket = std::os::raw::c_int;
+
+pub fn get_log_level() -> LogLevel {
+    let verbose = std::env::var("OPENVAS_LIBSSH_DEBUG")
+        .map(|x| x.parse::<i32>().unwrap_or_default())
+        .unwrap_or(0);
+
+    match verbose {
+        0 => LogLevel::NoLogging,
+        1 => LogLevel::Warning,
+        2 => LogLevel::Protocol,
+        3 => LogLevel::Packet,
+        _ => LogLevel::Functions,
+    }
+}
+
+fn to_comma_separated_string<T: AsRef<str>>(items: &[T]) -> Option<String> {
+    if items.is_empty() {
+        None
+    } else {
+        Some(
+            items
+                .iter()
+                .map(|name| name.as_ref().to_string())
+                .collect::<Vec<_>>()
+                .join(","),
+        )
+    }
```

Comment:
nit: I think reduce would have been a more approachable choice:
```
    items
        .iter()
        .map(|name| name.as_ref().to_string())
        .reduce(|acc, name| acc + ", " + &name)
```

Reply:
Note that we have to do
```rust
    items
        .iter()
        .map(|name| name.as_ref().to_string())
        .reduce(|acc, name| acc + ", " + &name)
        .unwrap_or("".to_string())
```

to deal with empty iterators, so I am not sure how much cleaner it is. Alternatively, `fold` with `""` as starting value works too. A third alternative is `itertools::join(..., ",")` which I think is the cleanest and avoids collecting altogether (although I think internally it probably just does something similar to your reduce call anyways).